CRM/Contribute/Form/CancelSubscription.php

```diff
@@ -220,14 +220,16 @@ public function postProcess() {
 
     if (CRM_Utils_Array::value('send_cancel_request', $params) == 1) {
       $cancelParams = ['subscriptionId' => $this->_subscriptionDetails->subscription_id];
-      $this->_paymentProcessorObj->setContributionRecurID($this->contributionRecurID);
-      $cancelSubscription = $this->_paymentProcessorObj->cancelSubscription($message, $cancelParams);
+      try {
+        $this->_paymentProcessorObj->setContributionRecurID($this->contributionRecurID);
+        $cancelSubscription = $this->_paymentProcessorObj->cancelSubscription($message, $cancelParams);
+      }
+      catch (\Civi\Payment\Exception\PaymentProcessorException $e) {
```

Comment:
@artfulrobot I meant, and I think that we should also catch the CRM_Core_Error here

Reply:
hmm not sure about that - there is no good reason for a Core_error to be available for catching there.

Reply:
@eileenmcnaughton I was hoping we could get rid of the `if (is_a('CRM_Core_Error'` statements below - but I may not be fully understanding how they are working?

Reply:
@artfulrobot You are right, of course :-)